### PR TITLE
IDEMPIERE-5595: Allow to deactivate negate option for "Chosen Multipl…

### PR DESCRIPTION
**Pull request by nmicoud**
…e Selection" components in processes

https://idempiere.atlassian.net/browse/IDEMPIERE-5595

# Pull Request Checklist

## Checklist:

- [X] My code follows the [code guidelines](https://wiki.idempiere.org/en/Contributing_to_Trunk) of this project
- [X] My code follows the best practices of this project
- [X] I have performed a self-review of my own code
- [X] My code is easy to understand and review. 
- [ ] In hard-to-understand areas, I have commented my code.
- [X] My changes generate no new warnings
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added unit tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings
- [ ] I have made corresponding changes to the documentation as follows:
- - [ ] New feature (non-breaking change which adds functionality): I have created the New Feature page in the project wiki explaining the functionality and how to use it. If relevant, I have committed sample data to the core seed to have usable examples in GardenWorld.
- - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected): I have documented the change in a clear way that everyone in the community can understand the impact of the change.
- - [ ] Improvement (improves and existing functionality): This documentation is needed if the improvement changes the way the user interacts with the system or the outcome of a process/task changes. If it is just, for instance, a performance improvement, documentation might not be needed. 
- [ ] The changed/added documentation is in the project wiki (not privately-hosted pdf files or links pointing to a company website) and is complete and self-explanatory.

### CODE REVIEW

**Comment by CarlosRuiz-globalqss** [comment]
@nmicoud - can you please solve the conflict on X_AD_Process_Para and move the scripts to iD10 folder

**Comment by nmicoud** [comment]
Hi @CarlosRuiz-globalqss 
Scripts moved to iD10
For the conflict, I guess it was because my local sources were a little bit outdated.
I reintegrated changes manually, not sure if it was what you expected.
If not, I'll close the current PR and recreate a new one

**Comment by nmicoud** [comment]
Ah, just discovered the Resolve Conflict button.
Does it worked ?

**Review by CarlosRuiz-globalqss — CHANGES_REQUESTED** [review]
Please move the new field "Show Negate Button" close to the Reference field (maybe below?)

Please also add the following UPDATE to the migration scripts:

```sql
UPDATE AD_Process_Para SET IsShowNegateButton='Y' WHERE AD_Process_ID IN 
(SELECT AD_Process_ID FROM AD_Process WHERE ClassName IS NULL AND JasperReport IS null)
```
(otherwise you're disabling this feature completely)

File: org.adempiere.base/src/org/compiere/model/MProcessPara.java [hunk 1]
```diff
@@ -351,7 +350,13 @@ && getAD_Element_ID() != 0 && isCentrallyMaintained())
 				LogicEvaluator.validate(getDisplayLogic());
 			}
 		}
-		
+
+		if (newRecord && DisplayType.isChosenMultipleSelection(getAD_Reference_ID())) {
+			MProcess p = MProcess.get(getAD_Process_ID());
+			if (Util.isEmpty(p.getClassname()) && Util.isEmpty(p.getJasperReport()))
```
Comment (CarlosRuiz-globalqss) [inline org.adempiere.base/src/org/compiere/model/MProcessPara.java:356]: Here you must check also that ProcedureName is empty

Not sure about the impact on Workflow and Special Form - must those be empty too?

File: migration/iD10/postgresql/202303020910_IDEMPIERE-5595.sql [hunk 2]
```diff
@@ -105,3 +105,73 @@ UPDATE AD_Field SET IsDisplayed='Y', SeqNo=330, XPosition=5,Updated=TO_TIMESTAMP
...
+
+-- Mar 2, 2023, 4:16:34 PM CET
+UPDATE AD_Field SET SeqNo=320,Updated=TO_TIMESTAMP('2023-03-02 16:16:34','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=10 WHERE AD_Field_ID=204372
+;
+
+-- Mar 2, 2023, 4:16:34 PM CET
+UPDATE AD_Field SET SeqNo=330,Updated=TO_TIMESTAMP('2023-03-02 16:16:34','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=10 WHERE AD_Field_ID=206410
+;
+
+UPDATE AD_Process_Para SET IsShowNegateButton='Y' WHERE AD_Process_ID IN
```
Comment (CarlosRuiz-globalqss) [inline migration/iD10/postgresql/202303020910_IDEMPIERE-5595.sql:176]: @nmicoud - you forgot to add this one to the oracle script too

Comment (nmicoud) [reply migration/iD10/postgresql/202303020910_IDEMPIERE-5595.sql:176]: hi @CarlosRuiz-globalqss ,
I was just wondering why the file was not saved in notepadd++ :)
Thanks,

**Review by CarlosRuiz-globalqss — APPROVED** [review]
tested

**Pull request merged**